src/colorizer/ColorizeCanvas.ts

```diff
@@ -555,11 +562,9 @@ export default class ColorizeCanvas {
    */
   updateRamp(): void {
     if (this.featureName && this.dataset?.isFeatureCategorical(this.featureName)) {
-      const palette = this.categoricalPalette;
-      const colorRamp = new ColorRamp(palette.colorStops);
-      this.setUniform("colorRamp", colorRamp.texture);
+      this.setUniform("colorRamp", this.categoricalPalette.texture);
```

Comment:
By saving `categoricalPalette` as a `ColorRamp`, we don't need to regenerate a `ColorPalette` on every render.